sphere-models/src/it/java/io/sphere/sdk/cartdiscounts/commands/CartDiscountInActualCartTest.java

```diff
@@ -25,21 +32,46 @@ public void createACartDiscountAndGetTheDiscountedValueFromACart() throws Except
+                final MonetaryAmount oldLineItemTotalPrice = MoneyImpl.ofCents(3702, EUR);
+                assertThat(undiscountedLineItem.getTotalPrice()).isEqualTo(oldLineItemTotalPrice);
+
+
                 final LocalizedString name = en("di");
                 final CartDiscountPredicate cartPredicate = CartDiscountPredicate.of(format("customer.id = \"%s\"", customer.getId()));
-                final AbsoluteCartDiscountValue value = CartDiscountValue.ofAbsolute(EURO_1);
+                final MonetaryAmount discountAmount = EURO_1;
+                final AbsoluteCartDiscountValue value = CartDiscountValue.ofAbsolute(discountAmount);
                 final CartDiscountDraft discountDraft = CartDiscountDraftBuilder.of(name, cartPredicate, value, LineItemsTarget.ofAll(), randomSortOrder(), false)
                         .build();
                 final CartDiscount cartDiscount = execute(CartDiscountCreateCommand.of(discountDraft));
                 final Cart cartIncludingDiscount = execute(CartUpdateCommand.of(cartWithCustomer, Recalculate.of()));
 
-                assertThat(cartIncludingDiscount.getTotalPrice().getNumber().doubleValueExact()).
-                        isEqualTo(cart.getTotalPrice().subtract(EURO_1).getNumber().doubleValueExact(), Offset.offset(0.05));
-                final DiscountedLineItemPrice discountedPrice = cartIncludingDiscount.getLineItems().get(0).getDiscountedPrice();
-                final Reference<CartDiscount> discount = discountedPrice.getIncludedDiscounts().get(0).getDiscount();
-                assertThat(discount.referencesSameResource(cartDiscount)).isTrue();
+                assertThat(cartIncludingDiscount.getTotalPrice()).
+                        isEqualTo(cart.getTotalPrice().subtract(discountAmount));
+
+
+                final LineItem lineItemWithDiscount = cartIncludingDiscount.getLineItems().get(0);
+                final List<DiscountedLineItemPriceForQuantity> discountedPricePerQuantity =
+                        lineItemWithDiscount.getDiscountedPricePerQuantity()
+                                .stream()
+                                .sorted(Comparator.comparing(x -> x.getQuantity()))
+                                .collect(toList());
+                assertThat(discountedPricePerQuantity).hasSize(2);
+
+                assertThat(discountedPricePerQuantity.get(0).getQuantity()).isEqualTo(1);
+                assertThat(discountedPricePerQuantity.get(0).getDiscountedPrice().getValue()).isEqualTo(MoneyImpl.ofCents(1200, EUR));
+                assertThat(discountedPricePerQuantity.get(0).getDiscountedPrice().getIncludedDiscounts()).hasSize(1);
+                assertThat(discountedPricePerQuantity.get(0).getDiscountedPrice().getIncludedDiscounts().get(0).getDiscount()).isEqualTo(cartDiscount.toReference());
+                assertThat(discountedPricePerQuantity.get(0).getDiscountedPrice().getIncludedDiscounts().get(0).getDiscountedAmount()).isEqualTo(MoneyImpl.ofCents(34, EUR));
```

Comment:
I would extract the line items discounted prices into variables to simplify the assertions:
```
discountedPricePerQuantity.get(0).getDiscountedPrice()
```